library/core/src/option.rs

```diff
@@ -671,8 +671,8 @@ impl<T> Option<T> {
     #[rustc_const_stable(feature = "const_option_basics", since = "1.48.0")]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub const fn as_ref(&self) -> Option<&T> {
-        match *self {
-            Some(ref x) => Some(x),
+        match self {
+            Some(x) => Some(x),
```

Comment:
This seems harder to read than before, it's not obvious that the type is being coerced.